src/main/java/org/terasology/launcher/gui/javafx/SettingsController.java

```diff
@@ -202,8 +204,15 @@ protected void saveSettingsAction(ActionEvent event) {
             launcherSettings.setUserGameParameters(userGameParametersField.getText());
         }
 
-        // store changed settings
+        // validate and store changed settings
         try {
+            if (validator.validateMaxHeapSize(launcherSettings)) {
+                updateMaxHeapSizeBox();
```

Comment:
I would merge this part as is, but I'm still wondering whether it would make sense to pass the `validator` to the `update(…)` method, like so:
```java
updateMaxHeapSizeBox(validator);
```
Does this bloat the implementation of `updateMaxHeapSizeBox(…)` too much? Currently, it's just a thin update function for the UI…